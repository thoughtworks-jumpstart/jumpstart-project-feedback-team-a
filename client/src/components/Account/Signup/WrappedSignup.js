import React from "react";
import Signup from "./Signup";
import * as auth from "../../../actions/auth";
import { object, instanceOf } from "prop-types";
import { withCookies, Cookies } from "react-cookie";

import { ProviderContext, subscribe } from "react-contextual";
import {
  mapMessageContextToProps,
  mapSessionContextToProps,
  messageContextPropType,
  sessionContextPropType
} from "../../context_helper";

export class WrappedSignup extends React.Component {
  static propTypes = {
    history: object.isRequired,
    cookies: instanceOf(Cookies).isRequired,
    ...messageContextPropType,
    ...sessionContextPropType
  };

  constructor(props) {
    super(props);
    this.state = { name: "", email: "", password: "", confirm: "" };
  }

  componentWillUnmount() {
    this.props.messageContext.clearMessages();
  }

  componentDidMount() {}

  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  handleSignup(event) {
    event.preventDefault();
    auth.signup({
      name: this.state.name,
      email: this.state.email,
      password: this.state.password,
      confirm: this.state.confirm,
      history: this.props.history,
      cookies: this.props.cookies,
      messageContext: this.props.messageContext,
      sessionContext: this.props.sessionContext
    });
  }

  render() {
    return (
      <Signup
        {...this.props}
        {...this.state}
        handleSignup={this.handleSignup.bind(this)}
        handleChange={this.handleChange.bind(this)}
      />
    );
  }
}

const mapContextToProps = context => {
  return {
    ...mapSessionContextToProps(context),
    ...mapMessageContextToProps(context)
  };
};

export default subscribe(ProviderContext, mapContextToProps)(
  withCookies(WrappedSignup)
);
