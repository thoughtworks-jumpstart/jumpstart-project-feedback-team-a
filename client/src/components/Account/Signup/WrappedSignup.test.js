import { WrappedSignup } from "./WrappedSignup";
import React from "react";
import { shallow } from "enzyme";
import { Cookies } from "react-cookie";
import { instanceOf } from "prop-types";

describe("WrappedSignup", () => {
  it("should should have the following structure", () => {
    const props = {
      history: {},
      cookies: instanceOf(Cookies),
      messageContext: { messages: {} },
      sessionContext: { session: {} }
    };
    const wrapper = shallow(<WrappedSignup {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
