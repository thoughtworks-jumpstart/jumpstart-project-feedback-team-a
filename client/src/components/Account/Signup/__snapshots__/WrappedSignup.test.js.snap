// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WrappedSignup should should have the following structure 1`] = `
<Signup
  confirm=""
  cookies={[Function]}
  email=""
  handleChange={[Function]}
  handleSignup={[Function]}
  history={Object {}}
  messageContext={
    Object {
      "messages": Object {},
    }
  }
  name=""
  password=""
  sessionContext={
    Object {
      "session": Object {},
    }
  }
/>
`;
